### ISSUE TITLE
example 79 is invalid because is uses 'width', 'height' property on Image (they are only defined as in Domain of Link)

### ISSUE THREAD

**Issue by gobengo** [w3c/activitystreams#522]
Please Indicate One: 
 * [x] Editorial
 * [ ] Question
 * [ ] Feedback
 * [ ] Blocking Issue
 * [ ] Non-Blocking Issue

Please Describe the Issue:
* via https://github.com/gobengo/activitystreams2-spec-scraped/issues/15
* the vocabulary sepc defines 'width' and 'height' properties as having a Domain of Link
* but ex 79 and 80 use these properties on an Object type ('Image'), which is not in the domain of these properties

recommendations:
* definitely: fix example 79 and 80 to either not use width/height on Image, or else to have the Image.url value be a Link, and put the height/width on there
* maybe: consider expanding the domain of 'width', 'height' to include `Image`. While this may be tempting... I don't think it's strictly necessary. It's nice to be able to describe an Image that might have multiple URLs to embed with different sizes, e.g. to render the Image + many Links into a [<picture>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture)

**Comment by sebilasse** [comment]
IMO it should be valid for both.
There are many many reasons and the main one is that we have responsive images and a srcset needing it anyway. Also sharing another article https://www.smashingmagazine.com/2020/03/setting-height-width-images-important-again/

**Comment by evanp** [comment]
We already have an erratum that says example 80 is technically incorrect and recommends using a Link.

We have several other remediations we can take:

1. Update the errata to note that example 79 has an identical problem with example 80.

2. A Erratum that says that the domain of `width` and `height` were intended to also extend to an `Object` or something more specific like `Image` or `Video`. It's been pointed out that in the wild this is largely assumed to be the case already.

3. An extension or other documentation that defines how and when to apply `width` and `height` to `Image`, `Video`, or other types.

**Comment by evanp** [comment]
I'm writing my book, and it sets my teeth on edge to say that you can put a width/height value on a Link but not an Image or a Video. We need to fix this in the next version, since it sucks.

Right now, the best option I can think of for this is using the Dublin Core terms for width and height:

```json
{
  "@context": [
    "https://www.w3.org/ns/activitystreams",
    {"dcterms": "https://purl.org/dc/terms/"}
  ],
  "type": "Image",
  "url": "http://example.com/image.jpg",
  "dcterms:width": "1024px",
  "dcterms:height": "768px"
}
```

**Comment by gobengo** [comment]
> but not an Image or a Video

@evanp et al, which normative language in the text do you see as prohibiting this?

I don't see it rfc2119-prohibited anywhere in the spec text, nor do I see any rfc2119-requirements about how it can be used.

While I filed this issue as an editorial one curiosity, I no longer find it too pressing since I've been doing more requirements parsing/tesitng. A reading of the testable statements in the activity-vocabulary doesn't seem to prohibit using it as in the example above. I encourage people to use whatever properties they want wherever they want and only be constrained by explicit requirements/prohibitions in the spec.

> We need to fix this in the next version, since it sucks.

As with all errata, this can be remedied by a [candidate correction](https://www.w3.org/2023/Process-20230612/#candidate-amendments) to include `Image` (or `Object`) alongside `Link` in the informative descriptions of the domains of `width` and `height` to avoid the confusion. As part of the required 'wide review' of such resolutions, I would be happy support such a proposal.

> Right now, the best option I can think of for this is using the Dublin Core terms for width and height:
This *is* a requirement in activitystreams-vocabulary: "However, to avoid possible interoperability issues, implementations MUST avoid using extension types or properties that unduly overlap with or duplicate the extended vocabulary defined here."

imo just use `width` `height` from as2 in as2 docs and on any resource it adds value to. It's unlikely anything breaks due to having it on non-Link-typed objects (as evidenced by Mastodon using it on `Documents` and things working fine).

**Comment by csarven** [comment]
@evanp , regarding your DC Terms width/height use, AFAICT those properties do not exist. Did I misinterpret what you're saying or did you mean something else?

**Comment by evanp** [comment]
@gobengo the Domain of `width` and `height` is `Link` only.

**Comment by evanp** [comment]
@csarven you're correct; dcterms doesn't have width and height. I changed it to Schema.org.

**Comment by johnspurlock** [comment]
Yes, let's get this fixed - the Domain of `width` and `height` should be `Object` | `Link`

The document itself already includes examples of `width` and `height` on `Image`s (which are not `Link`s)

https://www.w3.org/TR/activitystreams-vocabulary/#ex77-jsonld
<img width="544" alt="image" src="https://github.com/user-attachments/assets/adba5b52-6d18-4e00-85b3-6455bd89f2d4">

**Comment by sebilasse** [comment]
addendum:
I do find Example 79 and 80 (multiple sizes) misleading **for practical reasons** …

<img src="https://github.com/user-attachments/assets/0f40fa95-5dad-42ea-98dc-24b198c685ba" alt="AS Specification Example 82" width="64%"/>

---
_This is a justification why `width` and `height` SHOULD be used on `Link`;_
If a user wants to use 2 different icons and my app would want to show both:
How should my app know, if these are 2 different icons?
It is impossible (except you read every incoming media and create a content hash to compare).
I do not want to show all icon sizes at once.

It is pretty obvious that we need `width` and `height` for the `srcset` which is an HTML attribute (for `Link`s and not an HTML tag.
It would be clear and obvious for these:

**Example 79**
```json
{
  "@context": "https://www.w3.org/ns/activitystreams",
  "summary": "A simple note",
  "type": "Note",
  "content": "This is all there is.",
  "icon": {
    "type": "Image",
    "name": "Note icon",
    "url": [
      {
        "type": ["Link"],
        "href": "http://example.org/note.png",
        "width": 16,
        "height": 16
      }
    ]
  }
}
```

**Example 80**
```json
{
  "@context": "https://www.w3.org/ns/activitystreams",
  "summary": "A simple note",
  "type": "Note",
  "content": "A simple note",
  "icon": [
    {
      "type": "Image",
      "summary": "Note Icon",
      "url": [
        {
          "type": ["Link"],
          "summary": "Note (16x16)",
          "href": "http://example.org/note16.png",
          "width": 16,
          "height": 16
        },
        {
          "type": ["Link"],
          "summary": "Note (32x32)",
          "href": "http://example.org/note32.png",
          "width": 32,
          "height": 32
        }
      ]
    },
    {
      "type": "Image",
      "summary": "Funny British Humour Note Icon",
      "url": [
        {
          "type": ["Link"],
          "summary": "MontyNote (16x16)",
          "href": "http://example.org/montyNote16.png",
          "width": 16,
          "height": 16
        },
        {
          "type": ["Link"],
          "summary": "MontyNote (32x32)",
          "href": "http://example.org/montyNote32.png",
          "width": 32,
          "height": 32
        }
      ]
    }
  ]

}
```
